src/prefect/tasks/github/repos.py

```diff
@@ -70,3 +70,85 @@ def run(self, repo: str = None, info_keys: List[str] = None) -> None:
+            be provided to the `run` method.  Defaults to "master".
+        - branch_name (str, optional): the name of the new branch; can also be provided to the `run` method
+        - token_secret (str, optional): the name of the Prefect Secret containing your GitHub Access Token;
+            defaults to "GITHUB_ACCESS_TOKEN"
+        - **kwargs (Any, optional): additional keyword arguments to pass to the standard Task init method
+    """
+
+    def __init__(
+        self,
+        repo: str = None,
+        base: str = "master",
+        branch_name: str = None,
+        token_secret: str = "GITHUB_ACCESS_TOKEN",
+        **kwargs: Any
+    ):
+        self.repo = repo
+        self.base = base
+        self.branch_name = branch_name
+        self.token_secret = token_secret
+        super().__init__(**kwargs)
+
+    @defaults_from_attrs("repo", "base", "branch_name")
+    def run(self, repo: str = None, base: str = None, branch_name: str = None) -> dict:
+        """
+        Run method for this Task. Invoked by calling this Task after initialization within a Flow context,
+        or by using `Task.bind`.
+
+        Args:
+            - repo (str, optional): the name of the repository to open the issue in; must be provided in the
+                form `organization/repo_name`; defaults to the one provided at initialization
+            - base (str, optional): the name of the branch you want to branch off; if not provided here,
+                defaults to the one set at initialization
+            - branch_name (str, optional): the name of the new branch; if not provided here, defaults to
+                the one set at initialization
+
+        Raises:
+            - ValueError: if a `repo` was never provided, or if the base branch wasn't found
+            - HTTPError: if the GET request returns a non-200 status code
+
+        Returns:
+            - dict: dictionary of the response (includes commit hash, etc.)
+        """
+        if repo is None:
+            raise ValueError("A GitHub repository must be provided.")
+
+        ## prepare the request
+        token = Secret(self.token_secret).get()
+        url = "https://api.github.com/repos/{}/git/refs".format(repo)
+        headers = {
+            "AUTHORIZATION": "token {}".format(token),
+            "Accept": "application/vnd.github.v3+json",
+        }
+
+        ## gather branch information
+        resp = requests.get(url + "/heads", headers=headers)
+        resp.raise_for_status()
+        branch_data = resp.json()
+
+        commit_sha = None
+        for branch in branch_data:
+            if branch.get("ref") == "refs/heads/{}".format(base):
+                commit_sha = branch.get("object", {}).get("sha")
+                break
+
+        if commit_sha is None:
+            raise ValueError("Base branch {} not found.".format(base))
+
+        ## create new branch
+        new_branch = {"ref": "refs/heads/{}".format(branch_name), "sha": commit_sha}
```

Comment:
What if `branch_name` is None?

Reply:
Oh that's a good call out - it would create a new branch called `"None"`.  That is almost certainly not what users would want, so I'll raise a `ValueError` for that too.

Reply:
Updated!